pkg/resource/stack/deployment.go

```diff
@@ -478,49 +470,28 @@ func SerializePropertyValue(ctx context.Context, prop resource.PropertyValue, en
 		if showSecrets {
 			secret.Plaintext = plaintext
 		} else {
-			// If the encrypter is a cachingCrypter, call through its encryptSecret method, which will look for a matching
-			// *resource.Secret + plaintext in its cache in order to avoid re-encrypting the value.
-			var ciphertext string
-			if cachingCrypter, ok := enc.(*cachingSecretsManager); ok {
-				ciphertext, err = cachingCrypter.encryptSecret(ctx, prop.SecretValue(), plaintext)
+			// If the encrypter is a batchEncrypter, use the Enqueue method to asynchronously encrypt the secret value.
+			if batchEncrypter, ok := enc.(BatchEncrypter); ok {
+				err = batchEncrypter.Enqueue(ctx, prop.SecretValue(), plaintext, &secret)
+				if err != nil {
+					return nil, fmt.Errorf("enqueuing secret value for encryption: %w", err)
+				}
 			} else {
-				ciphertext, err = enc.EncryptValue(ctx, plaintext)
-			}
-			if err != nil {
-				return nil, fmt.Errorf("failed to encrypt secret value: %w", err)
+				ciphertext, err := enc.EncryptValue(ctx, plaintext)
+				if err != nil {
+					return nil, fmt.Errorf("failed to encrypt secret value: %w", err)
+				}
+				secret.Ciphertext = ciphertext
 			}
-			contract.AssertNoErrorf(err, "marshalling underlying secret value to JSON")
-
-			secret.Ciphertext = ciphertext
 		}
 
-		return secret, nil
+		return &secret, nil
```

Comment:
This is subtle -- the return type hasn't changed, but this is now a pointer. How does this continue to "just work"?